mklock.js

```diff
@@ -0,0 +1,84 @@
+const assert        = require("assert")
+const fs            = require("fs")
+const lockfile      = require("@yarnpkg/lockfile")
+const semver        = require("semver")
+const ssri          = require("ssri")
+
+// Usage: node mklock.js package-lock.json package.json yarn.lock integrities.json
+
+let pkgLockFile     = process.argv[2]
+let pkgFile         = process.argv[3]
+let yarnLockFile    = process.argv[4]
+let intFile         = process.argv[5]
+
+let pkgJson         = JSON.parse(fs.readFileSync(pkgFile, "utf8"))
+let yarnJson        = lockfile.parse(fs.readFileSync(yarnLockFile, "utf8")).object
+let integrities     = intFile ? JSON.parse(fs.readFileSync(intFile)) : {}
+
+let pkgDeps         = { ...(pkgJson.devDependencies || {}),
+                        ...(pkgJson.dependencies    || {}) }
+
+function objKeys(o) {
+  let keys = Object.keys(o)
+  keys.sort()
+  return keys
+}
+
+function splitNameVsn(key) {
+  // foo@vsn or @foo/bar@vsn
+  if (key[0] == "@") {
+    let [name, vsn] = key.slice(1).split("@")
+    return ["@"+name, vsn]
+  } else {
+    return key.split("@")
+  }
+}
+
+function addDeps(obj, pkg, vsn, seen) {
+  let pkgvsn            = pkg + "@" + vsn
```

Comment:
how about more consts?

Reply:
Yeah. Thanks for the reminder. `let` feels more natural if you're used to lisp, but here `const` is indeed better.